cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java

```diff
@@ -369,7 +381,11 @@ public void close() throws IOException {
   private FuzzyRowFilter createFuzzyRowFilter(FactScan scan, byte[] startRow) {
     // we need to always use a fuzzy row filter as it is the only one to do the matching of values
 
-    byte[] fuzzyRowMask = codec.createFuzzyRowMask(scan.getTagValues(), scan.getMeasureName());
+    // if we are querying only one metric, we will use fixed metricName for filter,
+    // if there are no metrics or >1 metrics to query we use `ANY` fuzzy filter.
+    String measureName =
+      (scan.getMeasureNames() != null && scan.getMeasureNames().size() == 1) ? scan.getMeasureNames().get(0) : null;
```

Comment:
Maybe in FactScan never have null? i.e. in constructor if user passes null - set empty collection? In FactScan and in QubeQueries it would simplify a lot of client code..